Added tag support.

LongForWisdom opened the pull request:
> Added tag fields to the preamble section of the general template.
> Added tag fields to the preamble section of the technical template.
> Added tag fields to the preamble section of each MIP.
> Added appropriate tags to each MIP in the tags fields of the preamble sections.

CPSTL commented:
@LongForWisdom If we do add tag support, I think we need to discuss the initial tags more thoroughly. Additionally,  we will need to formally define tags, how to use of them, the initial tags themselves, etc. This would need to be included in MIP0 or at least in the README.

LongForWisdom commented:
Aight. So actually the confusion here may be my fault. I intended to setup the templates such that the 'Tags' field had the <set by MIP Editor> field on. Buuuut it looks like I forgot to do that.

If the MIP Editor sets these then I don't think they need to be formally defined, I don't think we need to detail their use, I think they can purely be used by MIP Editors to aid discoverability in front-ends. I did consider adding a statement somewhere that tags should be used for that purpose, but there isn't really a good component in which to do that. I guess it could go in the MIP Editors roles / responsibilities.

Regarding the initial tag set, I mostly just went with what made sense to me. I'll provide justifications here for the tags used:
**core:** Defines MIPs that define core governance processes. Highlights MIPs that have a direct bearing on the governance process.
**'Role' tag (mip-editor, governance-facilitator, various domain teams, etc):** Highlights MIPs that have a direct bearing on how that role operates. 
**Process:** Anything with a Process Component in it, useful if someone wants to understand the different ways they can interact with the protocol.
**problem-space:** Admittedly this one is more speculative, anything that relates to the problem-space definition. Changes, a new problem-space, whatever. Might be less useful.
**collateral-onboarding:** Anything related to the onboarding of new collateral. Useful if you want to onboard new collateral.
**oracles:** Anything related to oracles, if you want to become an oracle or join/create an oracle team.
**risk:** Anything under the domain of risk, useful if you want to understand what risk teams do.
**interim:** Anything that is expected to at some point become obsolete. Useful if you want to understand what is impermanent.
**governance-cycle:** Anything relating to the governance cycle. May be less useful.
**emergency:** Anything related to emergency handling.
**application:** Anything that involves applying for something. Collateral, Oracles, etc. Useful if you want to apply for things? May not be necessary.
**active-list:** Any list of ratified/active roles/entities. Domain teams, General Risk Models, etc.

LongForWisdom commented:
Made some additional minor fixes during the merge.
- Changed MIP5 such that it's type is 'General' (no sub proposals / process components)
- Fixed whitespace formatting in MIP10 to match the rest.
